### PR TITLE
Update distributions to Python 3.

### PR DESCRIPTION
**Pull request by emilyfertig**
All unit tests pass, except those that depend on `goftests`, which is not yet updated to py3.

I had to make some hacky changes to `setup.py` to get this running (e.g. hard-coded local file paths), which aren't included in this PR. If someone wants to update `setup.py`, I'm happy to share what I did, if it's useful.

### CODE REVIEW

**Comment by jfinkels** [comment]
I updated goftests so that it should work with recent versions of numpy now, and there is a CI test for Python 3.10 https://github.com/posterior/goftests/pull/24

File: distributions/tests/test_model_flavors.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
 
 def test_model():
     for name in MODULES:
-        yield _test_model, name
+        _test_model(name)
```
Comment (fritzo) [inline distributions/tests/test_model_flavors.py:42]: I'd like to eventually switch from nose to pytest and make this a pytest fixture, but that's out of scope for this PR...

**Review by fritzo — COMMENTED** [review]
Changes look good to me, but I haven't tested locally.

WDYT of replacing .travis.yml with a .github/workflows/ci.yml as @jfinkels did in https://github.com/posterior/goftests/pull/23? Directly in this PR? Do we need to release goftests before that will pass?

**Comment by fritzo** [comment]
@emilyfertig I've released goftests 0.3.0 and parsable 0.3.0 which should now support python 3. Could you test again? LMK if you need any else.

**Comment by emilyfertig** [comment]
Thanks for releasing those! All `loom` unit tests are passing now, but I ran the pubmed example and got the following. It looks like another place in `parsable` that needs updating for py3?

```
Traceback (most recent call last):
  File "/usr/local/google/home/emilyaf/posterior_loom/loom/examples/pubmed/main.py", line 158, in <module>
    parsable.dispatch()
  File "/usr/local/google/home/emilyaf/.virtualenvs/loom/lib/python3.11/site-packages/parsable.py", line 125, in dispatch
    inspect.formatargspec(*inspect.getargspec(fun)),
    ^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'inspect' has no attribute 'formatargspec'. Did you mean: 'formatargvalues'?
```

**Comment by fritzo** [comment]
Thanks, looks like I need stronger tests for parsable. Let me fix that and cut another release.

**Comment by emilyfertig** [comment]
(I should have put that last comment on the `loom` PR discussion, not this one.)

`distributions` tests are all passing now, after increasing `SAMPLE_COUNT` to get a few of the GOF tests to pass.

I looked at replacing .travis.yml with .github/workflows/ci.yml, but I didn't do it because wasn't clear to me how to translate this piece of travis.yaml:
```
env:
  - FLAVOR=_cc
  - FLAVOR=_cc DISTRIBUTIONS_USE_PROTOBUF=1
  - FLAVOR=_cc_examples
  - FLAVOR=_cy
install:
  - make install$FLAVOR
```

Also, `setup.py` is likely still broken (I made some local changes to it to get the pip package to build, and when I remove those changes it now appears to work, but I suspect I'm not building it clean.)

Besides travis.yaml and setup.py, I think this is now mergeable.

**Review by fritzo — APPROVED** [review]

**Comment by fritzo** [comment]
@emilyfertig I can take a look at the travis -> github/workflows migration, and look at setup.py as part of that. If I have trouble I may ask for help if you recall what you did.

**Comment by emilyfertig** [comment]
Thanks, @fritzo ! My changes to setup.py are here: https://github.com/posterior/distributions/commit/8074e53fd64382b05a515758b17b1deee8f30c3f The commented-out portions are what initially seemed necessary to get it working, but now I'm not sure.

**Pull request merged**